### PR TITLE
Change TabBar to BottomNavigationBar

### PR DESCRIPTION
**Pull request by ferrarafer**
This option works using IndexedStack.

### CODE REVIEW

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
This looks good, but I would like the change tab to be `model.changeTab(index)`. Which means `MainView` would need a model. Also, someone using the architecture might want a tab view or nav bar view. So maybe creating two files might be the best option.

**Comment by ferrarafer** [comment]
> This looks good, but I would like the change tab to be `model.changeTab(index)`. Which means `MainView` would need a model. Also, someone using the architecture might want a tab view or nav bar view. So maybe creating two files might be the best option.

What do you mean with someone might want a TabBarView?

File: lib/ui/views/main_view.dart [hunk 1]
```diff
@@ -12,42 +14,34 @@ class MainView extends StatefulWidget {
 }
 
 class _MainViewState extends State<MainView> {
```
Comment (VictorUvarov) [inline lib/ui/views/main_view.dart]: The main view doesn't need to be stateful anymore.

File: lib/core/ui_models/views/main_model.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import 'package:provider_start/core/enums/view_state.dart';
+import 'package:provider_start/core/ui_models/base_model.dart';
+
+class MainModel extends BaseModel {
+  int _currentIndex = 0;
+  int get currentIndex => _currentIndex;
+
+  void changeTab(index) {
```
Comment (VictorUvarov) [inline lib/core/ui_models/views/main_model.dart]: change to 
```dart 
int index
```

File: lib/core/ui_models/views/main_model.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    setState(ViewState.Busy);
+    _currentIndex = index;
+    setState(ViewState.Idle);
```
Comment (VictorUvarov) [inline lib/core/ui_models/views/main_model.dart]: don't need to setState(...) twice, just  call once with
```dart
notifyListeners();
```

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
few changes

**Review by VictorUvarov — APPROVED** [review]

**Pull request merged**